### PR TITLE
[exporter/datadogexporter] Inheriting exporthelper config in

### PR DESCRIPTION
**Pull request by grzn**
**Description:**
In this PR, I've extended the datadog exporter to config to inherit the export helper config like many of the stock exports do (e.g. oltpexporter)

**Link to tracking Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/5420

### CODE REVIEW

**Review by mx-psi — CHANGES_REQUESTED** [review]
Thanks a lot for doing this @grzn! The way you added it to the configuration looks good but you will need to fix some unit tests in `factory_test.go` by adding the default configuration to the new configuration structs.

You can check the Github Actions output or run `make test` inside the `exporter/datadogexporter` folder to run them locally :)

**Comment by grzn** [comment]
Sure thing, updating the tests now.

File: exporter/datadogexporter/config/config.go
Comment (mx-psi) [inline exporter/datadogexporter/config/config.go:26]: This is no longer needed after the merge
```suggestion
```

**Comment by grzn** [comment]
Let me make sure that the tests pass

**Comment by grzn** [comment]
The metadata only test fails with the queue settings enabled. 
I'll probably have time to dive into this tomorrow.

**Comment by mx-psi** [comment]
> The metadata only test fails with the queue settings enabled.

That's interesting. We don't really need to add the queue settings to the config for that specific test to pass, but it would be nice to know why the timeout is happening. My guess would be that you need to call the `Start` method for the exporter since the queue exporter bits need something to start.

**Comment by grzn** [comment]
My thoughts exactly. I'll update once I iron that issue out.

**Comment by grzn** [comment]
Tests now pass on my machine.

**Comment by grzn** [comment]
Got any pointers for me to fix the integration tests?

**Review by mx-psi — APPROVED** [review]
Looks good to me, thanks again for doing this!

**Comment by mx-psi** [comment]
> Got any pointers for me to fix the integration tests?

Looks like a flake, I re-ran the tests and hopefully they will pass now

**Comment by grzn** [comment]
Same failure

**Comment by grzn** [comment]
It keeps failing on the memory consumption test, it is legit to increase it to 86mb?

**Comment by mx-psi** [comment]
Load tests do not use the datadogexporter code and are flaky so it is fine to ignore them. I marked as `ready-to-merge`, so this will probably land in the next contrib release

**Comment by mx-psi** [comment]
> It keeps failing on the memory consumption test, it is okay if I increase it 86mb?

Sorry, missed this. Feel free to open a separate PR for this if you want to, it's a known issue unrelated to your change but it would be nice to fix it

**Comment by grzn** [comment]
Okay I'll do that.

I can't merge it because of the failure, so how is it going to get into the next release?

**Pull request merged**
